rosidl_typesupport_c/src/type_support_dispatch.hpp

```diff
@@ -63,25 +62,30 @@ get_typesupport_handle_function(
           fprintf(stderr, "Failed to find library '%s'\n", library_name);
           return nullptr;
         }
-        lib = new Poco::SharedLibrary(library_path);
+
+        try {
+          lib = new rcpputils::SharedLibrary(library_path.c_str());
+        } catch (const std::runtime_error & e) {
+          throw std::runtime_error("Could not load library " + std::string(e.what()));
+        } catch (const std::bad_alloc & e) {
+          throw std::bad_alloc();
```

Comment:
This is basically doing nothing. Please handle the bad alloc case in the same way as the runtime error.